### DISCUSSION TITLE
Toggle mini.animate on/off

### DISCUSSION THREAD

**Discussion by V4G4X** [nvim-mini/mini.nvim#925] [Ideas]
I LOVE the animations of mini.animate, but I have learned that the animations may become extremely choppy/slow when in conjunction with other plugins. 

For example, when I'm reviewing merge requests using the split diff windows set up by https://github.com/sindrets/diffview.nvim. 
Scrolling these windows with animations ON  is quite slow and helps you practice your patience. 

It would be convenient to be able to either:
 - Toggle Animations as and when you want.
 - Disable animations opened by some specific plugins. 

What do you guys think? 

P.S. In case you think I should speed up my animations. I already did.  
Here's my config: https://github.com/V4G4X/nvim_config/blob/main/lua/core/plugin_configs/animate.lua

**Comment by echasnovski** [comment]
Thanks for the suggestion!

> It would be convenient to be able to either:
> 
>     * Toggle Animations as and when you want.
> 
>     * Disable animations opened by some specific plugins.

Both are already possible:

- Toggling specific animation can be done with the following code: `MiniAnimate.config.scroll.enable = not MiniAnimate.config.scroll.enable` (replace `scroll` with animation type you'd like to toggle). This, of course, can be done as mapping with something like this:

    ```lua
    local rhs = '<Cmd>lua MiniAnimate.config.scroll.enable = not MiniAnimate.config.scroll.enable<CR>'
    vim.keymap.set('n', '<Leader>ts', rhs, { desc = 'Toggle scroll animation' })
     ```

    Beware that it is expected that effect might be visible *after* the next scroll (as cache needs to be updated).

- Disabling animations locally is possible [per buffer using `vim.b.minianimate_config` buffer-local variable](https://github.com/echasnovski/mini.nvim/blob/335fdd637e905c5cdb8d7e7a9602e38cea9193af/doc/mini-animate.txt#L98). Usually it can be set in some `FileType` autocommand or 'after/ftplugin' directory. To disable scroll animation use this code: `vim.b.minianimate_config = { scroll = { enable = false } }`.
------

> Scrolling these windows with animations ON is quite slow and helps you practice your patience.

I like the delicate phrasing here. Will definitely use this in future :)

**Reply by V4G4X** [reply]
Thanks a lot. I love this!
